PROD-679 - Change the default group and profile images

milanhirapra opened the pull request:
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Fixes #3191 .

### How to test the changes in this Pull Request:

1.
2.
3.

### Proof Screenshots or Video

<!-- Add proof video or screenshots of what is fixed or added -->

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

**Platform API Issue/PR**
https://github.com/buddyboss/buddyboss-platform-api/pull/378

**Theme Issue/PR :**
https://github.com/buddyboss/buddyboss-theme/pull/1860

**App Plugin Issue/PR**
https://github.com/buddyboss/buddyboss-app/pull/2603
https://github.com/buddyboss/buddyboss-app/pull/2635

**App Plugin Issue/PR 2**
https://github.com/buddyboss/buddyboss-app/issues/2600
https://github.com/buddyboss/buddyboss-app/pull/2603

**App Native Issue**
https://github.com/buddyboss/buddyboss-app-native/issues/1851

**Jira issue** : [PROD-679]

[PROD-679]: https://buddyboss.atlassian.net/browse/PROD-679?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

sourabhmatolia commented:
@mikejyoung 

Moving the conversation to PR now since PR exists.

Check Video sharing more concerns where we need confirmation:
https://www.loom.com/share/89e981bdc6bc481d84fcb9c0ae8d2d54

Thanks

sourabhmatolia commented:
@milanhirapra 

Quick feedback on Custom Profile Avatar upload option:
https://www.loom.com/share/774c5e3fd4574932baff8bb95321b2a8

sourabhmatolia commented:
**UPDATE**

Mikey updated the Issue description and Mockup that clears all concerns shared so far. Thanks

eisenwasser commented:
@milanhirapra @sourabhmatolia 

We need to update the default color option in the theme for cover photos. 

Go to BuddyBoss > Theme Options > Styling, then search for "Cover Image Background"

Subtext needs to change to:

> Only visible if you have enabled cover images for [member profiles](#) and [groups](#).

- [Member Profiles] links to BuddyBoss > Settings > Profiles
- [Groups] links to BuddyBoss > Settings > Profiles

Then, this needs to be updated to WORK. Currently in my testing it does not do anything.

Then, we need to update the default color scheme so this option uses the background color from our Figma file, so our default BB cover image looks good.

Video explanation - https://www.loom.com/share/09f6ba3bd8104748a603e75a1b2cae36

eisenwasser commented:
@milanhirapra @sourabhmatolia 

The options in Profile settings and Group are not following the order of the wireframe:

https://www.loom.com/share/5b6c443f52e94548946df99f47f75a4a

- BuddyBoss needs to come before WordPress in dropdown
- BuddyBoss needs to come before None in image selections

sourabhmatolia commented:
@milanhirapra 

For the Member directory page we anyways don't show the cover image in that layout no matter Theme active or no, so nothing changes there.

For the Group directory page, previously default cover image was coming from Theme options, so maybe we decided not to show the cover image when BB theme not active.
But now we migrated that option from Theme to Platform, So I would recommend adding cover image layout in Group listing without BB Theme also.

Thanks

milanhirapra commented:
@sourabhmatolia @mikejyoung 

We've added a new metbox for group images and added a tutorial button, so we need a tutorial link to link a tutorial button.
Please check here: https://nimb.ws/dNmgMG

mikejyoung commented:
@milanhirapra 
We'll create the tutorial and link to it prior to launch.

chetansatasiya commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+		<input id="bp-default-profile-avatar-buddyboss" name="bp-default-profile-avatar-type" type="radio" value="buddyboss" <?php checked( bb_get_default_profile_avatar_type(), 'buddyboss' ); ?> />
+		<label for="bp-default-profile-avatar-buddyboss">
+			<div class="img-block">	
+				<img src="<?php echo buddypress()->plugin_url . 'bp-core/images/bb-group-avatar-buddyboss.jpg'; ?>" />
```
```suggestion
				<img src="<?php echo esc_url( buddypress()->plugin_url . 'bp-core/images/bb-group-avatar-buddyboss.jpg' ); ?>" />
```

chetansatasiya commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+		<input id="bp-default-profile-avatar-legacy" name="bp-default-profile-avatar-type" type="radio" value="legacy" <?php checked( bb_get_default_profile_avatar_type(), 'legacy' ); ?> />
+		<label for="bp-default-profile-avatar-legacy">
+			<div class="img-block">
+				<img src="<?php echo buddypress()->plugin_url . 'bp-core/images/bb-profile-avatar-legacy.jpg'; ?>" />
```
```suggestion
				<img src="<?php echo esc_url( buddypress()->plugin_url . 'bp-core/images/bb-profile-avatar-legacy.jpg' ); ?>" />
```

chetansatasiya commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+		<input id="bp-default-profile-avatar-custom" name="bp-default-profile-avatar-type" type="radio" value="custom" <?php checked( bb_get_default_profile_avatar_type(), 'custom' ); ?> />
+		<label for="bp-default-profile-avatar-custom">
+			<div class="img-block">
+				<img src="<?php echo buddypress()->plugin_url . 'bp-core/images/bb-avatar-custom.jpg'; ?>" />
```
```suggestion
				<img src="<?php echo esc_url( buddypress()->plugin_url . 'bp-core/images/bb-avatar-custom.jpg' ); ?>" />
```

chetansatasiya commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+
+	<div class="bb-default-custom-upload-file custom-profile-avatar custom-profile-group-avatar">
+		<div class="bb-upload-container">
+			<img src="<?php echo $custom_avatar_url; ?>" class="bb-upload-preview user-custom-avatar" data-default="<?php echo $placeholder_avatar_url; ?>" style="<?php echo $hide_show_style; ?>">
```
```suggestion
			<img src="<?php echo esc_url( $custom_avatar_url ); ?>" class="bb-upload-preview user-custom-avatar" data-default="<?php echo esc_url( $placeholder_avatar_url ); ?>" style="<?php echo esc_attr( $hide_show_style ); ?>">
```

chetansatasiya commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+	<div class="bb-default-custom-upload-file custom-profile-avatar custom-profile-group-avatar">
+		<div class="bb-upload-container">
+			<img src="<?php echo $custom_avatar_url; ?>" class="bb-upload-preview user-custom-avatar" data-default="<?php echo $placeholder_avatar_url; ?>" style="<?php echo $hide_show_style; ?>">
+			<input type="hidden" name="bp-default-custom-profile-avatar" id="bp-default-user-custom-avatar" value="<?php echo bb_get_default_custom_upload_profile_avatar(); ?>">
```
```suggestion
			<input type="hidden" name="bp-default-custom-profile-avatar" id="bp-default-user-custom-avatar" value="<?php echo esc_url( bb_get_default_custom_upload_profile_avatar() ); ?>">
```

chetansatasiya commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+		</div>
+		<div class="bb-img-button-wrap">
+			<a href="#TB_inline?width=800px&height=400px&inlineId=bp-xprofile-avatar-editor" class="button button-large thickbox bp-xprofile-avatar-user-edit"><?php esc_html_e( 'Upload', 'buddyboss' ); ?></a>
+			<a href="#" class="delete button button-large bb-img-remove-button bp-delete-custom-avatar bp-delete-custom-profile-avatar" style="<?php echo $hide_show_style; ?>"><?php esc_html_e( 'Remove', 'buddyboss' ); ?></a>
```
```suggestion
			<a href="#" class="delete button button-large bb-img-remove-button bp-delete-custom-avatar bp-delete-custom-profile-avatar" style="<?php echo esc_attr( $hide_show_style ); ?>"><?php esc_html_e( 'Remove', 'buddyboss' ); ?></a>
```

chetansatasiya commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+		<input id="bp-default-profile-cover-default" name="bp-default-profile-cover-type" type="radio" value="buddyboss" <?php checked( bb_get_default_profile_cover_type(), 'buddyboss' ); ?> />
+		<label for="bp-default-profile-cover-default">
+			<div class="img-block">
+				<img src="<?php echo buddypress()->plugin_url . 'bp-core/images/bb-cover-buddyboss.jpg'; ?>" />
```
```suggestion
				<img src="<?php echo esc_url( buddypress()->plugin_url . 'bp-core/images/bb-cover-buddyboss.jpg' ); ?>" />
```

chetansatasiya reviewed (changes requested):
@milanhirapra Many places no escaping functions are used run the PHPCS and all the changed the file in this PR and fix all of the issues.

KartikSuthar commented:
@milanhirapra I have a few concerns over here.

1. When moving the branch the first time: Buddyboss shows default selected instead of Legacy: http://prntscr.com/23zw8pt
2. Default preview color is wrong: http://prntscr.com/23zwcnw should be `#e2e9ef` which is applied on the front end: http://prntscr.com/23zwflw
3. UI issue with learndash active: http://prntscr.com/23zxabu
4. I think we should auto close the Thickbox after 2/5 second of success upload: http://prntscr.com/23zxo04
5. When changing the custom avatar it will not instance affect the preview frame: http://prntscr.com/23zy5jc
When click on the default selection it was working.
6. Platform current branch, theme release branch as per the instruction we should need to follow the cover image background from the theme, In backend frame it works but not in frontend: http://prntscr.com/23zzeaz
7. We should need to hide this app link until we can not get the updated version of the app plugin: http://prntscr.com/24001mo otherwise user can not find the fields: http://prntscr.com/2400686
8. When uploading landscape image to the uploader it breaks: http://prntscr.com/2401891
9. All above issue with the group image as well.

CC: @chetansatasiya @sourabhmatolia @mikejyoung

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -389,7 +389,7 @@ function bp_admin_setting_callback_avatar_uploads() {
 
 	<input id="bp-disable-avatar-uploads" name="bp-disable-avatar-uploads" type="checkbox" value="1" <?php checked( ! bp_disable_avatar_uploads( false ) ); ?> />
-	<label for="bp-disable-avatar-uploads"><?php _e( 'Allow members to upload photos for profile avatars', 'buddyboss' ); ?></label>
+	<label for="bp-disable-avatar-uploads"><?php _e( 'Allow members to upload photos a profile avatars', 'buddyboss' ); ?></label>
```
```suggestion
	<label for="bp-disable-avatar-uploads"><?php esc_html_e( 'Allow members to upload photos a profile avatars', 'buddyboss' ); ?></label>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
 	?>
 	<input id="bp-disable-cover-image-uploads" name="bp-disable-cover-image-uploads" type="checkbox" value="1" <?php checked( ! bp_disable_cover_image_uploads() ); ?> />
-	<label for="bp-disable-cover-image-uploads"><?php _e( 'Allow members to upload cover images', 'buddyboss' ); ?></label>
+	<label for="bp-disable-cover-image-uploads"><?php _e( 'Enable cover images for member profiles', 'buddyboss' ); ?></label>
```
```suggestion
	<label for="bp-disable-cover-image-uploads"><?php esc_html_e( 'Enable cover images for member profiles', 'buddyboss' ); ?></label>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
-	<label for="bp-disable-cover-image-uploads"><?php _e( 'Allow members to upload cover images', 'buddyboss' ); ?></label>
+	<label for="bp-disable-cover-image-uploads"><?php _e( 'Enable cover images for member profiles', 'buddyboss' ); ?></label>
+
+	<p class="description"><?php echo __( 'When enabled, members will be able to upload cover images in their profile settings.', 'buddyboss' ); ?></p>
```
```suggestion
	<p class="description"><?php esc_html_e( 'When enabled, members will be able to upload cover images in their profile settings.', 'buddyboss' ); ?></p>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+	?>
+	<div class="avatar-custom-input">
+		<select name="bp-profile-avatar-type" id="bp-profile-avatar-type">
+			<option value="buddyboss" <?php selected( bb_get_profile_avatar_type(), 'buddyboss' ); ?>><?php _e( 'BuddyBoss', 'buddyboss' ); ?></option>
```
```suggestion
			<option value="buddyboss" <?php selected( bb_get_profile_avatar_type(), 'buddyboss' ); ?>><?php esc_html_e( 'BuddyBoss', 'buddyboss' ); ?></option>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+	<div class="avatar-custom-input">
+		<select name="bp-profile-avatar-type" id="bp-profile-avatar-type">
+			<option value="buddyboss" <?php selected( bb_get_profile_avatar_type(), 'buddyboss' ); ?>><?php _e( 'BuddyBoss', 'buddyboss' ); ?></option>
+			<option value="wordpress" <?php selected( bb_get_profile_avatar_type(), 'wordpress' ); ?>><?php _e( 'WordPress', 'buddyboss' ); ?></option>
```
```suggestion
			<option value="wordpress" <?php selected( bb_get_profile_avatar_type(), 'wordpress' ); ?>><?php esc_html_e( 'WordPress', 'buddyboss' ); ?></option>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+			<div class="img-block">	
+				<img src="<?php echo buddypress()->plugin_url . 'bp-core/images/bb-group-avatar-buddyboss.jpg'; ?>" />
+			</div>
+			<span><?php _e( 'BuddyBoss', 'buddyboss' ); ?></span>
```
```suggestion
			<span><?php esc_html_e( 'BuddyBoss', 'buddyboss' ); ?></span>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+			<div class="img-block">
+				<img src="<?php echo buddypress()->plugin_url . 'bp-core/images/bb-profile-avatar-legacy.jpg'; ?>" />
+			</div>
+			<span><?php _e( 'Legacy', 'buddyboss' ); ?></span>
```
```suggestion
			<span><?php esc_html_e( 'Legacy', 'buddyboss' ); ?></span>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+			<div class="img-block">
+				<img src="<?php echo buddypress()->plugin_url . 'bp-core/images/bb-avatar-custom.jpg'; ?>" />
+			</div>
+			<span><?php _e( 'Custom', 'buddyboss' ); ?></span>
```
```suggestion
			<span><?php esc_html_e( 'Custom', 'buddyboss' ); ?></span>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+		</label>
+	</div>
+
+	<p class="description"><?php echo __( 'Select which image should be used for members who haven\'t uploaded a profile avatar.', 'buddyboss' ); ?></p>
```
```suggestion
	<p class="description"><?php esc_html_e( 'Select which image should be used for members who haven\'t uploaded a profile avatar.', 'buddyboss' ); ?></p>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+	<div class="bp-cover-image-status bb-custom-profile-group-avatar-feedback" style="display:none;">
+		<p id="bp-avatar-image-feedback" class="updated"></p>
+	</div>
+	<p class="description"><?php echo sprintf( __( 'Upload an image to be used as the default profile avatar. Recommended size is %1$spx by %2$spx.', 'buddyboss' ), bp_core_avatar_full_width(), bp_core_avatar_full_height() ); ?></p>
```
```suggestion
	<p class="description"><?php echo sprintf( esc_html__( 'Upload an image to be used as the default profile avatar. Recommended size is %1$spx by %2$spx.', 'buddyboss' ), bp_core_avatar_full_width(), bp_core_avatar_full_height() ); ?></p>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+			<div class="img-block">
+				<img src="<?php echo buddypress()->plugin_url . 'bp-core/images/bb-cover-buddyboss.jpg'; ?>" />
+			</div>
+			<span><?php _e( 'BuddyBoss', 'buddyboss' ); ?></span>
```
```suggestion
			<span><?php esc_html_e( 'BuddyBoss', 'buddyboss' ); ?></span>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+			<div class="img-block">
+				<img src="<?php echo buddypress()->plugin_url . 'bp-core/images/bb-cover-none.jpg'; ?>" />
+			</div>
+			<span><?php _e( 'None', 'buddyboss' ); ?></span>
```
```suggestion
			<span><?php esc_html_e( 'None', 'buddyboss' ); ?></span>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+			<div class="img-block">
+				<img src="<?php echo buddypress()->plugin_url . 'bp-core/images/bb-cover-custom.jpg'; ?>" />
+			</div>
+			<span><?php _e( 'Custom', 'buddyboss' ); ?></span>
```
```suggestion
			<span><?php esc_html_e( 'Custom', 'buddyboss' ); ?></span>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+			<span><?php _e( 'Custom', 'buddyboss' ); ?></span>
+		</label>
+	</div>
+	<p class="description"><?php _e( 'Select which image should be used for members who haven\'t uploaded a profile cover image.', 'buddyboss' ); ?></p>
```
```suggestion
	<p class="description"><?php esc_html_e( 'Select which image should be used for members who haven\'t uploaded a profile cover image.', 'buddyboss' ); ?></p>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+	<p class="description">
+		<?php
+		echo sprintf(
+			__( 'Upload an image to be used as the default profile cover image. Recommended size is %1$spx by %2$spx.', 'buddyboss' ),
```
```suggestion
			esc_html__( 'Upload an image to be used as the default profile cover image. Recommended size is %1$spx by %2$spx.', 'buddyboss' ),
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+		<div class="preview-switcher-main">
+
+			<div class="button-group preview-switcher">
+				<a href="#web-preview" class="button button-large button-primary"><?php _e( 'Browser', 'buddyboss' ); ?></a>
```
```suggestion
				<a href="#web-preview" class="button button-large button-primary"><?php esc_html_e( 'Browser', 'buddyboss' ); ?></a>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-settings.php:
```diff
@@ -402,7 +402,240 @@ function bp_admin_setting_callback_avatar_uploads() {
+			<div class="button-group preview-switcher">
+				<a href="#web-preview" class="button button-large button-primary"><?php _e( 'Browser', 'buddyboss' ); ?></a>
+				<?php if ( $live_preview_settings['is_buddyboss_app_plugin_active'] ) : ?>
+					<a href="#app-preview" class="button button-large"><?php _e( 'App', 'buddyboss' ); ?></a>
```
```suggestion
					<a href="#app-preview" class="button button-large"><?php esc_html_e( 'App', 'buddyboss' ); ?></a>
```

KartikSuthar commented on src/bp-core/admin/bp-core-admin-functions.php:
```diff
@@ -3245,3 +3245,190 @@ function bp_core_get_moderation_admin_tabs( $active_tab = '' ) {
+
+	wp_send_json_success( $response );
+}
+add_action( 'wp_ajax_bb_profile_group_get_live_preview_urls', 'bb_profile_group_get_live_preview_urls' );
```
@milanhirapra I think we don't need this ajax to generate the preview box.
you can change the preview box using JS.

milanhirapra commented:
@mikejyoung 

Need transparent BuddyBoss cover images for web and app preview section.
Please check this screenshot: https://nimb.ws/KHrXBb

CC: @sourabhmatolia @firozwp

milanhirapra commented:
One thing has remained to replace buddyboss cover images with transparent buddyboss cover images for web and app preview section.
Here is the comment: https://github.com/buddyboss/buddyboss-platform/pull/3216#issuecomment-1000855315

CC: @sourabhmatolia @milanhirapra

sourabhmatolia commented:
@amincdev @jitendrabanjara1991 

**Initial feedback**

Check Detailed video on issues shared below:
https://www.loom.com/share/799b632fcac24d2baa7b3d4315f2a734

- Settings > Profile > Dropdown "WordPress" > Gravtar should not be disabled.

- Settings > Profile > dropdown BuddyBoss > Wrong English
"Allow members to upload photos a profile avatars"
Should be same as mockup

- Settings > Profile > dropdown BuddyBoss > why do we hide the default image option when " Allow members to upload photos a profile avatars" is not checked??
When this is checked then preview below also shows blank
  - We are not supposed to show blank in the frontend also.


- Frontend > When the wrong image size is uploaded then preview also looks wrong here:
https://prnt.sc/24r2b58
Same issue for the app also.

- Frontend > Preview looks bad if you look closely:
https://prnt.sc/24r2iem
Maybe we should not put borders for the image or color.


- Frontend > Upload Profile avatar popup height is not correct:
https://prnt.sc/24r3a69


- Clicking crop and success message flow are not correct.
On selecting image > Do not show "For best results, upload an image that is 300px by 300px or larger." notice.
  - Upload cover image > Delete > Do not show success message like:
Cover photo was deleted successfully.
Or on upload also:
Cover photo was uploaded successfully. For best results, upload an image that is 1950px by 450px or larger.
  - Explain upload and remove flow
(same for group)


- Settings > Groups > small glitch when refreshing here.
It shows hidden content for a second and then it hides.
Looks like we are hiding using js, instead of hide using CSS class.
  - Same for profile settings.

sourabhmatolia commented:
@milanhirapra 

**More feedback**

Check Video:
https://www.loom.com/share/cd75b909a82144948f508f5acec425dd

- Change image meta box order in Group settings
(reported by Michael)

- Save avatar/cover image settings on upload/delete on the fly.

Thanks.

mikejyoung commented:
@milanhirapra 

Some changes to "Group Avatar" settings

* Change back from "Upload Avatars" to "Group Avatars
* Change the checkbox text for "Group Avatars" to "Enable avatars for groups"
* Add "When enabled, group organizers will be able to upload avatars in the group's settings" helper text to "Group Avatars"
* The "Default Group Avatar" field only shows if "Group Avatars" is checked 

<img width="702" alt="Screen Shot 2021-12-29 at 7 50 25 AM" src="https://user-images.githubusercontent.com/75724470/147639559-31b13f5b-31e6-4b27-b79d-2b55bebdf569.png">

CC: @sourabhmatolia

mikejyoung commented:
@milanhirapra 

Added a notice in WordPress > Settings > Discussion

- If "Profile Avatars" is set to "BuddyBoss" show a notice in the "Avatars" section
  - Text is "Profile avatars are currently provided by the BuddyBoss Platform. To use the WordPress avatar system, change the Profile Avatars setting to "WordPress" in the Profile settings."
  - "Profile" links to BuddyBoss > Settings > Profiles

<img width="1383" alt="Screen Shot 2021-12-29 at 8 05 33 AM" src="https://user-images.githubusercontent.com/75724470/147640635-f2f0afd7-25bc-41d4-b083-4e479c75af65.png">

CC: @sourabhmatolia

mikejyoung commented:
@milanhirapra @amincdev 

Some style changes....

### Image Select Field Type

This should be update in all places using this field type (Profile Images, Group Images, On-Screen Notifications)

**Active Option**
`border: 2px solid #2271b1`

**Inactive Option**
`border: 2px solid #2271b1`

<img width="722" alt="Screen Shot 2021-12-29 at 10 27 05 AM" src="https://user-images.githubusercontent.com/75724470/147653155-527e410b-2b95-420a-8d6a-2b9f89c652e2.png">

### Upload Image Field Type

"Remove" text should be red
Notice style should match our standard styles for inside BuddyBoss metaboxes (@annrra did this for the app plugin - have these been implemented into Platform/Theme/Platform Pro @sourabhmatolia ?) 

<img width="794" alt="Screen Shot 2021-12-29 at 10 27 32 AM" src="https://user-images.githubusercontent.com/75724470/147653389-b393da43-3035-4b52-8e33-2929c27470d3.png">

<img width="1140" alt="Screen Shot 2021-12-29 at 10 34 11 AM" src="https://user-images.githubusercontent.com/75724470/147653510-7ef08acf-06ce-47e6-ba68-6196f6eb7cc9.png">

### Preview Cover Image


**Wrapper**

```
border: 1px solid #CCD0D4;
border-radius: 5px;
overflow: hidden;
```

**Profile Avatar**

``` border-radius: 5px; ```

<img width="731" alt="Screen Shot 2021-12-29 at 10 27 11 AM" src="https://user-images.githubusercontent.com/75724470/147653411-d5ed291b-2511-4622-9262-ad2c09a601a3.png">

amincdev commented:
> ### Image Select Field Type
> This should be update in all places using this field type (Profile Images, Group Images, On-Screen Notifications)
> 
> **Active Option** `border: 2px solid #2271b1`
> 
> **Inactive Option** `border: 2px solid #2271b1`
> 
> <img alt="Screen Shot 2021-12-29 at 10 27 05 AM" width="722" src="https://user-images.githubusercontent.com/75724470/147653155-527e410b-2b95-420a-8d6a-2b9f89c652e2.png">
> 

@mikejyoung Active and inactive state should have the same border width. Otherwise, there will be a layout shift.
So, please let me know whether should I add 1px width or 2px?

mikejyoung commented:
@amincdev 
Can you use box-shadow for active state border?

milanhirapra commented:
> @milanhirapra
> 
> Added a notice in WordPress > Settings > Discussion
> 
> * If "Profile Avatars" is set to "BuddyBoss" show a notice in the "Avatars" section
>   
>   * Text is "Profile avatars are currently provided by the BuddyBoss Platform. To use the WordPress avatar system, change the Profile Avatars setting to "WordPress" in the Profile settings."
>   * "Profile" links to BuddyBoss > Settings > Profiles
> 
> <img alt="Screen Shot 2021-12-29 at 8 05 33 AM" width="1383" src="https://user-images.githubusercontent.com/75724470/147640635-f2f0afd7-25bc-41d4-b083-4e479c75af65.png">
> 
> CC: @sourabhmatolia

@mikejyoung 
WordPress does not provide hook to customise or add text to the show avatar section in discussion page. 

CC: @sourabhmatolia @firozwp @KartikSuthar @chetansatasiya

sourabhmatolia commented:
@mikejyoung 

>Notice style should match our standard styles for inside BuddyBoss metaboxes (@annrra did this for the app plugin - have these been implemented into Platform/Theme/Platform Pro @sourabhmatolia ?)

I have this in my notes but didn't process that yet:
https://github.com/buddyboss/buddyboss-app/issues/1499
Will Prioritize.

@milanhirapra 

>WordPress does not provide hook to customise or add text to the show avatar section in discussion page.

No hook at all on the discussion page?
Are you sure?
Worst case, we have to use JS to add small notice.

amincdev commented:
> @amincdev Can you use box-shadow for active state border?

@mikejyoung  Yes, I have did the same way.

amincdev commented:
> <img alt="Screen Shot 2021-12-29 at 10 34 11 AM" width="1140" src="https://user-images.githubusercontent.com/75724470/147653510-7ef08acf-06ce-47e6-ba68-6196f6eb7cc9.png">
> 

@mikejyoung We have predefined styles for Notices: https://ibb.co/NSJ2GJq
Do I need to follow new styles shared above or I should just follow what we already have?

mikejyoung commented:
@amincdev 
How quickly can we update to new styles?
You can just pull them from buddyboss app plugin.

amincdev commented:
> @amincdev How quickly can we update to new styles? You can just pull them from buddyboss app plugin.

@mikejyoung Yes, I can quickly check the app plugin to pull those styles. Just to confirm this new design will be for this group cover images areas which we are working on right now, Not globally.

mikejyoung commented:
@amincdev 
The styles should be used globally for all notices appearing **inside** metaboxes in **any** of our product's (platform, theme, platform pro, app plugin) back-end settings.

CC: @sourabhmatolia

amincdev commented:
> @amincdev The styles should be used globally for all notices appearing **inside** metaboxes in **any** of our product's (platform, theme, platform pro, app plugin) back-end settings.
> 
> CC: @sourabhmatolia

As I discussed with @mikejyoung I will use the existing styling for notices: https://ibb.co/NSJ2GJq and we will restyle all notices across all our plugins and theme in another task. Which I will make sure to disucss with @sourabhmatolia and document it.

amincdev commented:
@mikejyoung All the Front-End related changes are done now.

mikejyoung commented:
@amincdev 
https://www.loom.com/share/3d3d69b848024349b45ec95b566839a2?from_recorder=1&focus_title=1

Give the avatar in the "Preview Cover Image" a background color when avatar is set to "Custom" so that it doesn't look weird while waiting to uploading a custom avatar image.

<img width="988" alt="Screen Shot 2021-12-29 at 2 39 36 PM" src="https://user-images.githubusercontent.com/75724470/147673690-eeda513f-ba8c-4056-a87f-8cd7f6b1404d.png">

amincdev commented:
> @amincdev 
> Give the avatar in the "Preview Cover Image" a background color when avatar is set to "Custom" so that it doesn't look weird while waiting to uploading a custom avatar image.

@mikejyoung Thank you for pointing this out. It is fixed now.

eisenwasser commented:
@amincdev @mikejyoung 

I think it's weird that when BB Platform is disabled, we still show a random "Browser" button - https://prnt.sc/256iuqv
Makes no sense, there should be no button at all.

Also the text doesn't make sense:
"On the web, the Cover Image Background color can be changed in the Theme Options."

Why do we need to specify "on the web" to a web only customer? 

I assume same issue for Groups as well.

eisenwasser commented:
@mikejyoung 

This text:
"In the app, it can be changed in the Color settings."

Color settings links to:
/wp-admin/admin.php?page=bbapp-appearance&setting=styling&screen=color-general

I cannot figure out which setting would change the cover image. Background Color? Alternate Background Color? 
We have no setting with a Cover Background type of label, and no profile preview for this color section, so this is not obvious at all. Just trial and error. We could make this much more clear by also stating which specific color option to change in the description.

eisenwasser commented:
@mikejyoung 
The preview image for "App" looks really terrible.
https://prnt.sc/256k2de

eisenwasser commented:
@milanhirapra 

Saving a custom group cover photo does not work at all.

https://www.loom.com/share/ee9ed402ac14400baa9df2689955517f

eisenwasser commented:
@milanhirapra 

Deleting a previously uploaded cover photo does not work at all.

https://www.loom.com/share/de126c7e2eb94e8face8fb17becaf4f6

eisenwasser commented:
@amincdev 

'Remove' button should have red text:

Current - https://prnt.sc/256lhxq
Should be - https://prnt.sc/256ljso

eisenwasser commented:
@milanhirapra 

If I set Group Cover photo to Custom, and do not upload any image, it saves it as Custom.
But, uses BuddyBoss on front end.
https://www.loom.com/share/a18ef1c4f34f476dbd57615bd9da26d1

Shouldn't it force it back to BuddyBoss in admin then?

eisenwasser commented:
@milanhirapra 

Every single time you save a custom avatar in backend we are generating an activity post that makes no sense that just says "changed their photo". - https://prnt.sc/256oaz3

No activity post should ever get generated from any of these backend settings.

https://www.loom.com/share/904db93eb7d5471ab94ea6518431b210

eisenwasser commented:
@amincdev Photo cropper does not match wireframe:

Looks like this (very ugly) - https://prnt.sc/256okf5
Wireframe - https://prnt.sc/256oocm

We need to update this cropper interface in following places:
- Profile avatar settings
- Group avatar settings
- Users > extended profile > change avatar settings

amincdev commented:
1. 
> I think it's weird that when BB App plugin is disabled, we still show a random "Browser" button - https://prnt.sc/256iuqv
Makes no sense, there should be no button at all.
> Also the text doesn't make sense:
> "On the web, the Cover Image Background color can be changed in the Theme Options."
> Why do we need to specify "on the web" to a web only customer?
> Same issue for Groups as well - https://prnt.sc/256j8t4

2.
> 'Remove' button should have red text:
> Current - https://prnt.sc/256lhxq
> Should be - https://prnt.sc/256ljso

3.
> Photo cropper does not match wireframe:
> Looks like this (very ugly) - https://prnt.sc/256okf5
> Wireframe - https://prnt.sc/256oocm
> We need to update this cropper interface in following places:
> BuddyBoss > Settings > Profiles > Profile Images > Default Profile Avatar > Custom
> BuddyBoss > Settings > Groups > Group Images > Default Group Avatar > Custom
> Users > Profile > Extended profile > Profile Photo settings -- https://prnt.sc/256p22b

@mikejyoung @eisenwasser This all points are done

milanhirapra commented:
> @milanhirapra
> 
> Saving a custom group cover photo does not work at all.
> 
> https://www.loom.com/share/ee9ed402ac14400baa9df2689955517f

=> It's done.

> @milanhirapra
> 
> Deleting a previously uploaded cover photo does not work at all.
> 
> https://www.loom.com/share/de126c7e2eb94e8face8fb17becaf4f6

=> We have changed custom avatar and cover upload directory so, old uploaded images will not delete.

> @milanhirapra
> 
> If I set Group Cover photo to Custom, and do not upload any image, it saves it as Custom. But, uses BuddyBoss on front end. https://www.loom.com/share/a18ef1c4f34f476dbd57615bd9da26d1
> 
> Shouldn't it force it back to BuddyBoss in admin then?

=> I don't replicate this. Please check here: https://vimeo.com/661112502/9567d92bff


> @milanhirapra
> 
> Every single time you save a custom avatar in backend we are generating an activity post that makes no sense that just says "changed their photo". - https://prnt.sc/256oaz3
> 
> No activity post should ever get generated from any of these backend settings.
> 
> https://www.loom.com/share/904db93eb7d5471ab94ea6518431b210

=> It's done.

mikejyoung commented:
@amincdev @milanhirapra 

> I think it's weird that when BB App plugin is disabled, we still show a random "Browser" button - https://prnt.sc/256iuqv
> Makes no sense, there should be no button at all.
> 
> Also the text doesn't make sense:
> "On the web, the Cover Image Background color can be changed in the Theme Options."
> 
> Why do we need to specify "on the web" to a web only customer?
> 
> Same issue for Groups as well - https://prnt.sc/256j8t4

When app is disabled:
- [ ] Remove the button switcher
- [ ] When theme is installed, change helper text to "The Cover Image Background color can be changed in the Theme Options."
- [ ] When theme is not installed, change helper text to "The Cover Image Background color can be changed with custom CSS."

> This text:
> "In the app, it can be changed in the Color settings."
> 
> [Color settings] links to:
> /wp-admin/admin.php?page=bbapp-appearance&setting=styling&screen=color-general
> 
> I cannot figure out which setting would change the cover image. Background Color? Alternate Background Color?
> We have no setting with a Cover Background type of label, and no profile preview for this color section, so this is not obvious at all. Just trial and error. We could make this much more clear by also stating which specific color option to change in the description.

@eisenwasser 
We are creating a new color option for "Cover Image Background" here:
https://github.com/buddyboss/buddyboss-app/issues/2600

These issue will be released at the same time.

mikejyoung commented:
@milanhirapra 

When app is enabled:

- [ ] When theme is installed, change helper text to "In a browser, the Cover Image Background color can be changed in the Theme Options. In the app, it can be changed in the Color settings."
- [ ]  When theme is not installed, change helper text to "In a browser, the Cover Image Background color can be changed with custom CSS. In the app, it can be changed in the Color settings."

chetansatasiya commented on src/bp-core/admin/bp-core-admin-functions.php:
```diff
@@ -2632,28 +2646,45 @@ function bp_core_admin_create_background_page() {
+				jQuery( document ).ready( function() {
+					var discussion_avatar_tbl = $( 'body.options-discussion-php #wpbody-content .wrap form table:eq(1)' );
+					if ( discussion_avatar_tbl.find( 'tr:eq(1)' ).hasClass( 'avatar-settings' ) ) {
+						discussion_avatar_tbl.prev().after( '<?php echo $avatar_notice_html; ?>' );
```
```suggestion
						discussion_avatar_tbl.prev().after( '<?php echo wp_kses( $avatar_notice_html ); ?>' );
```

chetansatasiya reviewed (changes requested):
need requested changes

mikejyoung commented:
@milanhirapra 

I've updated a number of the images used in the front-end and back-end for this issue.

Please replace all with those in this folder:
[Default profile and group images.zip](https://github.com/buddyboss/buddyboss-platform/files/7793194/Default.profile.and.group.images.zip)

CC: @sourabhmatolia

chetansatasiya commented on src/bp-core/bp-core-attachments.php:
```diff
@@ -1450,8 +1575,33 @@ function bp_attachments_cover_image_ajax_upload() {
+	 * @param string $type       The type of the attachment which is also the subdir where files are saved.
+	 *                           Defaults to 'cover-image'
+	 */
+	$cover_subdir = apply_filters( 'bp_attachments_get_attachment_sub_dir', $object_data['dir'] . '/' . $bp_params['item_id'] . '/cover-image', $object_data['dir'], $bp_params['item_id'], 'cover-image' );
```
```suggestion
	$cover_subdir = apply_filters( 'bb_attachments_get_attachment_sub_dir', $object_data['dir'] . '/' . $bp_params['item_id'] . '/cover-image', $object_data['dir'], $bp_params['item_id'], 'cover-image' );
```

Newly introduced filters should be start with `bb_` instead of `bp_`

chetansatasiya commented on src/bp-core/bp-core-attachments.php:
```diff
@@ -1450,8 +1575,33 @@ function bp_attachments_cover_image_ajax_upload() {
+	 * @param string $type       The type of the attachment which is also the subdir where files are saved.
+	 *                           Defaults to 'cover-image'
+	 */
+	$cover_dir = apply_filters( 'bp_attachments_get_attachment_dir', $cover_dir, $object_data['dir'], $bp_params['item_id'], 'cover-image' );
```
```suggestion
	$cover_dir = apply_filters( 'bb_attachments_get_attachment_dir', $cover_dir, $object_data['dir'], $bp_params['item_id'], 'cover-image' );
```

Newly introduced filters should be start with `bb_` instead of `bp_`

chetansatasiya commented on src/bp-core/bp-core-filters.php:
```diff
@@ -1643,3 +1643,490 @@ function bb_rest_decode_default_avatar_url( $gravatar ) {
+ *                                  Defaults to 'cover-image'.
+ * @return string Actual custom uploaded cover relative sub path.
+ */
+function bp_attachments_get_profile_group_attachment_sub_dir( $cover_sub_dir, $object_dir, $item_id, $type ) {
```
```suggestion
function bb_attachments_get_profile_group_attachment_sub_dir( $cover_sub_dir, $object_dir, $item_id, $type ) {
```

chetansatasiya commented on src/bp-core/bp-core-functions.php:
```diff
@@ -6033,3 +6034,566 @@ function bb_check_server_disabled_symlink() {
+ * @param string $component The BuddyPress component concerned ("xprofile" for user or "groups").
+ * @return array|bool An associative array containing the advised width and height for the cover photo. False if settings are empty.
+ */
+function bp_attachments_get_default_custom_cover_image_dimensions( $component = 'xprofile' ) {
```
```suggestion
function bb_attachments_get_default_custom_cover_image_dimensions( $component = 'xprofile' ) {
```

chetansatasiya commented on src/bp-core/bp-core-functions.php:
```diff
@@ -6033,3 +6034,566 @@ function bb_check_server_disabled_symlink() {
+}
+
+/**
+ * Has default custom upload group avatar?.
```
```suggestion
 * Has default custom upload group avatar?
```

eisenwasser commented:
@milanhirapra Still waiting on all images to be replaced, see Mikey's comment above.

Also in my testing on the dev site, this option seems to do nothing in latest app build - https://prnt.sc/25vhrsz
Is that not yet ready?

milanhirapra commented:
@eisenwasser 
I already replaced those images in this commit:: https://github.com/buddyboss/buddyboss-platform/pull/3216/commits/38e3a40a14c8cc6222cb11eb0b4da60bb689330e

milanhirapra commented:
@eisenwasser 
I replaced again all backend and frontend images, but I don't find any changes.

joeytoribio commented:
Hi @milanhirapra,

This has bee tested in the qa-ongoing branch with some minor issue listed below. Also you may refer on the googlesheet for all the Test Cases perform on this PR.

**Test Cases:** https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=1302572123&range=26:119 

**Failed TC:** 
1. TC38: Change existing Custom Cover Image - The existing position set by the user still not reset to default position
Video : https://www.loom.com/share/abe0d537648c4de183947279e1951347
TC: https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=1302572123&range=42:42
2. TC53: Change existing Custom Group Cover Image - The Current Position set by the user still not reset after change of custom background
Video:  https://www.loom.com/share/abe0d537648c4de183947279e1951347 
TC: https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=1302572123&range=57:57 
3. TC58: Upload Avatars is Disabled - The User is redirected to 404 page
Video: https://www.loom.com/share/532e27c02c014536bd8cc88f7c77b28c
TC: https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=1302572123&range=62:62
4. TC59: Delete Profile Photo - WordPress - The Profile Image at the top right is change to Gravatar icon instead of Wavatar
Video: https://www.loom.com/share/0d33e8d74ed141b0a9b3d4ca169c4123
TC: https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=1302572123&range=63:63
5. TC63: BuddyBoss Cover Image Reposition - There is a shadow showing when repositioning the buddyboss cover image
Video: https://www.loom.com/share/201768d9bc314605b6d3e83b61e0cb11  Time: 2:35 - 3:50
TC: https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=1302572123&range=67:67
6. TC64: Custom Cover Image - The Default Color set in the BuddyBoss > Theme Options > Styling > Cover Image Backgroud is showing quickly before the custom cover image will show 
Video: https://www.loom.com/share/201768d9bc314605b6d3e83b61e0cb11 Time: 11:50 - 12:20
TC: https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=1302572123&range=68:68
7. TC76: Delete Group Avatar - The user Profile is also changing and showing as the Default Group Profile Avatar
Video: https://www.loom.com/share/54b6448dddc94c078aa81b5e8be0319b
TC: https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=1302572123&range=80:80
8. TC77: Upload Group Avatar - the user Profile is also changing and showing the newly uploaded Image
Video: https://www.loom.com/share/e5f27a4fa0b2428ba9e4484f71b55973
TC: https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=1302572123&range=81:81
9. TC85: Group Cover Image Reposition - There is a shadow showing when repositioning the buddyboss cover image
Video: https://www.loom.com/share/dd44b7f65ea743da8e15ff894583648a
TC: https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=1302572123&range=89:89

Please let me know should you have any questions.

Thank you!

chetansatasiya reviewed (changes requested):
@milanhirapra create a deprecated file and add all the removed functions on that file. 

for e.g `bp_remove_avatar_settings_from_options_discussion_page` function

KartikSuthar commented on src/bp-core/bp-core-update.php:
```diff
@@ -1387,3 +1390,279 @@ function bb_update_to_1_8_1() {
+ */
+function bb_update_to_1_8_6() {
+	global $buddyboss_theme_options;
+
```
Please add those code snippets here:
```
// Do not ignore deprecated code for existing installs.
bp_add_option( '_bp_ignore_deprecated_code', false );
```

it will not skip the deprecated file load for existing installs.

chetansatasiya commented on src/bp-groups/bp-groups-filters.php:
```diff
@@ -215,29 +211,6 @@ function bp_groups_disable_at_mention_forums_notification_for_non_public_groups(
 	return $send;
 }
 
-/**
- * Use the mystery group avatar for groups.
- *
- * @since BuddyPress 2.6.0
- *
- * @param string $avatar Current avatar src.
- * @param array  $params Avatar params.
- * @return string
- */
-function bp_groups_default_avatar( $avatar, $params ) {
-	if ( isset( $params['object'] ) && 'group' === $params['object'] ) {
-		if ( isset( $params['type'] ) && 'thumb' === $params['type'] ) {
-			$file = 'mystery-group-50.png';
-		} else {
-			$file = 'mystery-group.png';
-		}
-
-		$avatar = buddypress()->plugin_url . "bp-core/images/$file";
-	}
-
-	return $avatar;
-}
-
```
This function should go to the deprecated file.

milanhirapra replied:
Done.

chetansatasiya reviewed (changes requested):
Need requested changes

chetansatasiya commented on src/bp-templates/bp-nouveau/buddypress/assets/emails/single-bp-email.php:
```diff
@@ -360,7 +360,7 @@
 								<td valign="middle" width="10%" style="vertical-align: middle;">
 									<a href="#" target="_blank">
-										<img alt="Avatar" src="<?php echo apply_filters( 'bp_email_default_avatar', buddypress()->plugin_url .'bp-core/images/mystery-man.jpg' ); ?>" width="45" height="45" style="border: 1px solid #b9babc; border-radius: 50%; vertical-align: middle;" />
+										<img alt="Avatar" src="<?php echo apply_filters( 'bp_email_default_avatar', bb_attachments_get_default_profile_group_avatar_image( array( 'object' => 'user' ) ) ); ?>" width="45" height="45" style="border: 1px solid #b9babc; border-radius: 50%; vertical-align: middle;" />
```
add `Avatar` to translation function

milanhirapra replied:
done
